http/transport/dump_round_tripper.go

```diff
@@ -114,12 +132,19 @@ func (l *DumpRoundTripper) RoundTrip(req *http.Request) (*http.Response, error)
 	if err != nil {
 		return resp, err
 	}
+
 	// response logging
 	if l.DumpResponse || l.DumpResponseHEX {
 		respDump, err := httputil.DumpResponse(resp, l.DumpBody)
 		if err != nil {
 			respDump = []byte(err.Error())
 		}
+
+		// in case a redactor is present, redact the content before logging
+		if redactor != nil {
```

Comment:
Minor: We could save a function call by checking whether `len(respDump) > 0` or something, to avoid calling it on an empty string

Reply:
the respDump is never nill (it is either the complete response or the error message). IMHO the check would not be working because of that.